README changes (#1656)
snowflakedb/snowpark-python: README.md

```bash
pip install snowflake-snowpark-python
```

> To use the [Snowpark pandas API](https://docs.snowflake.com/LIMITEDACCESS/snowpark-pandas), you can optionally install the following, which installs Modin in the same environment. The Snowpark pandas API provides a familiar interface for pandas users to query and process data directly in Snowflake.

Reviewer: ```suggestion
To use the [Snowpark pandas API](https://docs.snowflake.com/LIMITEDACCESS/snowpark-pandas), you can optionally install the following, which installs [Modin](https://github.com/modin-project/modin) in the same environment. The Snowpark pandas API provides a familiar interface for pandas users to query and process data directly in Snowflake.
```

Reviewer 2: [Modin] => [modin]

Author: Added link and changed capitalization

> To use the [Snowpark pandas API](https://docs.snowflake.com/LIMITEDACCESS/snowpark-pandas), you can optionally install the following, which installs Modin in the same environment. The Snowpark pandas API provides a familiar interface for pandas users to query and process data directly in Snowflake.

Reviewer 3: ```suggestion
To use the [Snowpark pandas API][Snowpark pandas developer guide], you can optionally install the following, which installs Modin in the same environment. The Snowpark pandas API provides a familiar interface for pandas users to query and process data directly in Snowflake.
```

> To use the [Snowpark pandas API][Snowpark pandas developer guide], you can optionally install the following, which installs [modin](https://github.com/modin-project/modin) in the same environment. The Snowpark pandas API provides a familiar interface for pandas users to query and process data directly in Snowflake.

Reviewer 3: ```suggestion
To use the [Snowpark pandas API][Snowpark pandas developer guide], you can optionally install the following, which installs [modin][modin] in the same environment. The Snowpark pandas API provides a familiar interface for pandas users to query and process data directly in Snowflake.
```

```bash
pip install "snowflake-snowpark-python[modin]"
```

> ### Create a session and use the Snowpark DataFrame API

Reviewer 4: since we are using Snowpark Python API for the other parts of doc, shall we keep this as Snowpark Python API

Author: Changing this back to Snowpark Python API.

> ### Create a session and use the Snowpark DataFrame APIs

Reviewer: ```suggestion
### Create a session and use the Snowpark DataFrame API
```

Author: Updated!

    logger.addHandler(ch)
```

## Reading and writing to pandas DataFrame

> Snowpark DataFrame API supports reading from and writing to a pandas DataFrame via the [to_pandas][to_pandas] and [write_pandas][write_pandas] commands. 

Reviewer 4: I think it would be good for us to mention that both Snowpark Python API and Snowpark pandas API provides to_pandas() to convert Snowpark Dataframe and Snowpark DataFrame to native pandas Dataframe. However, if the  user is a pure Snowpark user, in other words, have an environment without modin option installed, users will need to install the [pandas] option to get interaction with pandas object work.

Reviewer 4: we can also give a simple to_pandas example for snowpark pandas also, i believe we have example on our API doc

Author: Thanks! I've incorporated this in the updated README.

> Snowpark DataFrame API supports reading from and writing to a pandas DataFrame via the [to_pandas](https://docs.snowflake.com/developer-guide/snowpark/reference/python/latest/snowpark/api/snowflake.snowpark.DataFrame.to_pandas) and [write_pandas](https://docs.snowflake.com/developer-guide/snowpark/reference/python/latest/snowpark/api/snowflake.snowpark.Session.write_pandas) commands. 

Reviewer 3: Can we add these links to the bottom of the README that way we can just have [to_pandas] and [write_pandas] instead of these links for developer readability?

Author: Updated!

    logger.addHandler(ch)
```

## Reading and writing to pandas DataFrame

> Snowpark DataFrame API supports reading from and writing to a pandas DataFrame via the [to_pandas][to_pandas] and [write_pandas][write_pandas] commands. 

Reviewer 4: -> Snowpark DataFrame

Author: Updated

To use these operations, ensure that pandas is installed in the same environment. You can install pandas alongside Snowpark Python by executing the following command:
```bash
pip install "snowflake-snowpark-python[pandas]"
```
Once pandas is installed, you can convert between a Snowpark DataFrame and pandas DataFrame as follows: 
```python
df = session.create_dataframe([[1, 2], [3, 4]], schema=["a", "b"])
# Convert Snowpark DataFrame to pandas DataFrame
pandas_df = df.to_pandas() 
# Write pandas DataFrame to a Snowflake table and return Snowpark DataFrame
snowpark_df = session.write_pandas(pandas_df, "new_table", auto_create_table=True)
```

Snowpark pandas API also supports writing to pandas: 
```python

> import modin.pandas as pd

Reviewer 4: i think we can remove the import here like the python, because to have a complete example, it also needs the session. Since we already have an example above I think we should be fine here

Author: I removed the snowpark import but kept the modin one just so that it is clear this is a Snowpark pandas dataframe, since the API signature is the same as pandas.

## Reading and writing to pandas DataFrame

> Snowpark DataFrame API supports reading from and writing to a pandas DataFrame via the [to_pandas](to_pandas) and [write_pandas](write_pandas) commands. 

Reviewer 3: ```suggestion
Snowpark DataFrame API supports reading from and writing to a pandas DataFrame via the [to_pandas][to_pandas] and [write_pandas][write_pandas] commands. 
```

[snowflake lab sample code]: https://github.com/Snowflake-Labs/snowpark-python-demos
[samples]: https://github.com/snowflakedb/snowpark-python/blob/main/README.md#samples
[contributing]: https://github.com/snowflakedb/snowpark-python/blob/main/CONTRIBUTING.md
[to_pandas]: https://docs.snowflake.com/developer-guide/snowpark/reference/python/latest/snowpark/api/snowflake.snowpark.DataFrame.to_pandas

> [write_pandas]: https://docs.snowflake.com/developer-guide/snowpark/reference/python/latest/snowpark/api/snowflake.snowpark.Session.write_pandas

Reviewer 3: ```suggestion
[write_pandas]: https://docs.snowflake.com/developer-guide/snowpark/reference/python/latest/snowpark/api/snowflake.snowpark.Session.write_pandas
[modin]: https://github.com/modin-project/modin
```
